build-tools/packages/build-cli/test/commands/generate/buildVersion.test.ts

```diff
@@ -436,6 +440,81 @@ describe("generate:buildVersion", () => {
 			expect(ctx.stdout).to.contain("version=2.0.0-rc.7.0.0");
 			expect(ctx.stdout).to.contain("isLatest=false");
 		});
+
+	test
+		.env({
+			VERSION_BUILDNUMBER: "212045",
+			VERSION_TAGNAME: "client",
+			TEST_BUILD: "false",
+			VERSION_RELEASE: "release",
+			VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "2.0.0", "--tags", ...test_tags])
+		.it("2.0 version, release", (ctx) => {
+			expect(ctx.stdout).to.contain("version=2.0.0");
+			expect(ctx.stdout).to.contain("isLatest=false");
+		});
+
+	test
+		.env({
+			VERSION_BUILDNUMBER: "212045",
+			VERSION_TAGNAME: "client",
+			TEST_BUILD: "false",
+			VERSION_RELEASE: "prerelease",
+			VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "2.0.0", "--tags", ...test_tags])
+		.it("2.0 version, prerelease", (ctx) => {
+			expect(ctx.stdout).to.contain("version=2.0.0-212045");
+			expect(ctx.stdout).to.contain("isLatest=false");
+		});
+
+	test
+		.env({
+			VERSION_BUILDNUMBER: "212045",
+			VERSION_TAGNAME: "client",
+			TEST_BUILD: "true",
+			VERSION_RELEASE: "prerelease",
+			VERSION_INCLUDE_INTERNAL_VERSIONS: "False",
+		})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "2.0.0", "--tags", ...test_tags])
+		.it("2.0 version, test", (ctx) => {
+			expect(ctx.stdout).to.contain("version=0.0.0-212045-test");
+			expect(ctx.stdout).to.contain("isLatest=false");
+		});
+
+	test
+		.env({
+			VERSION_BUILDNUMBER: "212045",
+			VERSION_TAGNAME: "client",
+			TEST_BUILD: "false",
+			VERSION_RELEASE: "release",
+			VERSION_INCLUDE_INTERNAL_VERSIONS: "True",
+		})
+		.stdout()
+		.command(["generate:buildVersion", "--fileVersion", "2.1.2", "--tags", ...test_tags])
+		.it("2.1.2 version returns isLatest true", (ctx) => {
```

Comment:
For more clarity on what this case is testing? Then for the next one, "next unpublished minor version (2.2.0)...", and maybe have a test case for "next unpublished patch version on non-latest minor (2.0.2) returns isLatest false"?
```suggestion
		.it("next unpublished patch version on latest minor (2.1.2) returns isLatest true", (ctx) => {
```